tests/lib/linter.js

```diff
@@ -4480,6 +4480,16 @@ describe("Linter", () => {
                 assert.strictEqual(messages.length, 0);
             });
 
+            it("should not throw or report errors when the custom parser returns nested unrecognized operators (https://github.com/eslint/eslint/issues/10560)", () => {
+                const code = "null %% 'foo'";
```

Comment:
Although the `code` ultimately doesn't matter due to the nature of the parser fixture, could this line be modified to reflect the source code being simulated in the fixture, just for consistency? (`foo && bar ?? baz`)

Reply:
I manually modified the fixture so the operator was really `unknown`. I did it that way so when the `??` standardizes we don't have to change this test. Only add it to `isHandledLogicalOperator` and handle it in a couple of places.

Reply:
Apologies, I mistyped. I had meant to ask that this source code be used, to match the fixture: `foo && bar %% baz`

My issue is not that you're using an unknown operator, but rather that the `code` variable's value appears to have been copied, pasted, and not modified, compared to the unit test above this one.

Hope this helps and apologies again for the confusion!

Reply:
Oh, sorry! My brain parser only saw the change in the operator, not in the variables, haha. I'll change that now. Thanks!